presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1070,7 +1070,7 @@ private void prepareInsertExistingTable(HdfsContext context, TableAndMore tableA
         private void prepareDropPartition(SchemaTableName schemaTableName, List<String> partitionValues)
         {
             metastoreDeleteOperations.add(new IrreversibleMetastoreOperation(
-                    format("drop partition %s.%s %s", schemaTableName, schemaTableName.getTableName(), partitionValues),
+                    format("drop partition %s.%s %s", schemaTableName.getSchemaName(), schemaTableName.getTableName(), partitionValues),
```

Comment:
or just

```suggestion
                    format("drop partition %s %s", schemaTableName, partitionValues),
```

? @rschlussel ?

Reply:
that works, but I'm not sure if we want to rely on the toString method printing in this particular format. I mean it does right now, and it is very unlikely to change, but it could change without someone noticing that changing how this object gets printed would affect users.